src/main/java/org/havenapp/main/service/SignalSender.java

```diff
@@ -81,6 +90,56 @@ public void run() {
         });
     }
 
+    public void stopHeartbeatTimer ()
+    {
+        mCountdownTimer.cancel();
+        mCountdownTimer = null;
+        Log.d("HEARTBEAT TIMER", "Stopped" );
+    }
+
+    public void startHeartbeatTimer (int countMs)
+    {
+        //Set default to 5 minutes, catch dangerous Signal thresholds
+        if (countMs <= 10000)
+            countMs = 300000;
+
+        mCountdownTimer =  new CountDownTimer(countMs,1000) {
+
+            public void onTick(long millisUntilFinished) {
+                Log.d("HEARTBEAT TIMER"," seconds remaining: " + millisUntilFinished / 1000);
+            }
+
+            public void onFinish() {
+                Log.d("HEARTBEAT TIMER"," Done, update message sent!");
+                beatingHeart();
+                start();
+            }
+        }.start();
+    }
+
+    private void beatingHeart ()
```

Comment:
testNotification in ListViewActivity could accept a message string instead of adding all these lines.